### PR TITLE
Add Transmission Connector

### PR DESCRIPTION
**Pull request by itshoro**
This PR adds support for a *transmission*-connector.

It implements the **add**-, **get**- and  **remove**-actions, which are present in the legacy Yousei implementation.

There are some things I'm not completely sure about.

https://github.com/WiiPlayer2/Yousei/blob/02a117f47ebf14678b5e60841970290d3b1cf858/Yousei.Connectors/Transmission/AddAction.cs#L26

Is calling `ConfigureAwait(false)` necessary here?

https://github.com/WiiPlayer2/Yousei/blob/02a117f47ebf14678b5e60841970290d3b1cf858/Yousei.Connectors/Transmission/RemoveAction.cs#L22

This function isn't async, is there a better way to do this?

**At this point we would also still need to add the connector**, by callling `YouseiReloaded.Internal.ConnectorRegistry.Register(new TransmissionConnector())`, however I also am unsure on how and especially **where** I should call this.

### CODE REVIEW

**Comment by WiiPlayer2** [comment]
In library code `ConfigureAwait(false)` should almost always be used. I chose to ignore this for now as we are not using any synchronization context so the call doesn't matter.

If the transmission api method is not returning a task then there is nothing we can do about it. We'll just leave it be for now until there is an update which makes it return a task.

All static connectors should be registered in the concrete implementation namely inside the constructor of `YouseiReloaded.Internal.ConnectorRegistry`.

File: Yousei.Connectors/Transmission/GetAction.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        {
+            this.client = client;
+        }
+
+        protected override async Task Act(IFlowContext context, GetArguments arguments)
+        {
+            var fields = await arguments.Fields.Resolve<string[]>(context);
+            var ids = await arguments.Ids.Resolve<int[]>(context);
+
+            await client.TorrentGetAsync(fields, ids);
```
Comment (WiiPlayer2) [inline Yousei.Connectors/Transmission/GetAction.cs:22]: The information returned by this should be saved in the context.

File: Yousei.Connectors/Transmission/TransmissionConnector.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+using Yousei.Shared;
+
+namespace Yousei.Connectors.Transmission
+{
+    internal class TransmissionConnector : Connector<Unit>
+    {
+        private readonly TransmissionConnection connection;
+
+        public TransmissionConnector(Client client) : base("transmission")
+        {
```
Comment (WiiPlayer2) [inline Yousei.Connectors/Transmission/TransmissionConnector.cs:10]: The `TransmissionConnector` should not get a `Transmission.API.RPC.Client` and rather create it as part of its `GetConnection` method. The current implementation would disallow connections to different transmission servers and force the caller to connect to the server which goes against the architecture.
There should be configuration class containing all relevant connection arguments (api key etc.) and then those should be used in `GetConnection` to create a `TransmissionConnection` with a `Transmission.API.RPC.Client` with the correct api key etc.

**Review by WiiPlayer2 — CHANGES_REQUESTED** [review]

File: Yousei.Connectors/Transmission/GetAction.cs (hunk 1 above)
Comment (itshoro) [reply Yousei.Connectors/Transmission/GetAction.cs:22]: Should this also apply to `client.TorrentAddAsync()`?

Comment (WiiPlayer2) [reply Yousei.Connectors/Transmission/GetAction.cs:22]: If it returns a meaningful object (which it does if I recall correctly) then yes, this should also apply to `TorrentAddSync`.

File: Yousei.Connectors/Transmission/TransmissionConnector.cs (hunk 2 above)
Comment (WiiPlayer2) [reply Yousei.Connectors/Transmission/TransmissionConnector.cs:10]: This only allows to have at max a single connection to any transmission server which still goes against the architecture. I suggest using the newly introduced `SimpleConnector` as base class. You can take a look at `RssConnector` or `TelegramConnector` for examples on how to use it.

**Review by WiiPlayer2 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**